airflow/utils/docs.py

```diff
@@ -35,4 +40,20 @@ def get_docs_url(page: Optional[str] = None) -> str:
 
 def get_doc_url_for_provider(provider_name: str, provider_version: str) -> str:
     """Prepare link to Airflow Provider documentation."""
-    return f'https://airflow.apache.org/docs/{provider_name}/{provider_version}/'
+    try:
+        metadata_items = importlib_metadata.metadata(provider_name).get_all("Project-URL")
+        if isinstance(metadata_items, str):
```

Comment:
Warning! Rant: 

Who on earth came with the interface to importlib's metadata ????

1) when you do ['Project-URL'] - you get only the first matching entry in "project-URL"
2) when you do get_all() - you get all, normally when you get many, you get list, but when you ave one - you get string (WtF?)
3) Even if metadata in setup.py/cfg/pyproject.toml is structured, the "string" you get here is coma separated entry ("type, value")
4) The PEP 0621: https://peps.python.org/pep-0621/#example shows example of documentation metadata as:

```
[project.urls]
homepage = "example.com"
documentation = "readthedocs.org"
repository = "github.com"
changelog = "github.com/me/spam/blob/master/CHANGELOG.md"
```

Yet - you have to ue "Prioject-URL" and you get "Documentaiton" as output.
```

Reply:
Unless I am mistaken with any of those (@uranusjr ?) - this the shortest possible specification with the biggest possible number of traps set for the user of that API.

Reply:
You should report that as a bug. This does not sound right to me though, `get_all` is inherited from `email.message`, and AFAIK `get_all` always returns a list (unless `failobj` is returned, which is not the case here).

https://docs.python.org/3/library/email.message.html#email.message.EmailMessage.get_all

Reply:
Well this is what get_all () typing information tells.

I have not actually CHECKED if it works and I have no idea that get_all() in package_metadat is derived from EmailMessage (WHY??? O WHY???) .

I just follow the documentation of importlib_metadata: https://docs.python.org/3/library/importlib.metadata.html 
And it tells me absolutely nothing about what failobj is, and how it impacts the returned data. So I have to assume I can get a scalar.

But maybe I missed it :). Could you tell me how  I am supposed to know that get_all in PackageMetadata behaves like the one in Email Message ???? Did I miss it somewhere in the docs? 

From the source code it **looks like** the method can return scalar value. But I have no idea when as teh documentation nor source code does not tell me.

```
    def get_all(self, name: str, failobj: _T = ...) -> Union[List[Any], _T]:
        """
        Return all values associated with a possibly multi-valued key.
        """
```


So now we have 5th trap - undocumented behaviour of the most basic (and the only one that works) method to retrieve the metadata :). Honestly now the library is really at the top of my "trap list".  I think I know why people refrain from using metadata from packages.

Reply:
It inheriting from EmailMessage is an implementation detail, the goal is to parse a case-insensitive, multi-value mapping format. The documentation is indeed missing some flesh in exactly what type is returned by each function.

The non-stdlib backport importlib-metadata has better documentation on this:

https://importlib-metadata.readthedocs.io/en/latest/api.html#importlib_metadata.PackageMetadata.get_all

It’s far from comprehensive but you can see it promises to return either a `List[Any]`, or whatever type passed to `failobj`. Since you did not pass `failobj`, you should never get a str. So this sounds like a bug.